frc46_token/src/token/types.rs

```diff
@@ -129,7 +129,17 @@ pub struct MintReturn {
 }
 
 impl Cbor for MintReturn {}
-impl RecipientData for MintReturn {
+
+/// Intermediate data used by mint_return to construct the return data
+#[derive(Debug)]
+pub struct MintIntermediate {
+    /// Recipient address to use for querying balance
+    pub recipient: Address,
```

Comment:
Should we restrict visibility of these to the crate? I'm undecided, so ok to leave public for now.